patroni/api.py

```diff
@@ -110,6 +113,45 @@ def do_GET_patroni(self):
         response = self.get_postgresql_status(True)
         self._write_status_response(200, response)
 
+    def do_GET_config(self):
+        self._write_json_response(200, self.server.patroni.config.dynamic_configuration)
+
+    @staticmethod
+    def _patch_config(config, data):
+        for name, value in data.items():
+            if isinstance(value, dict) and name in config:
+                RestApiHandler._patch_config(config[name], value)
+            elif value is None:
+                config.pop(name, None)
+            else:
+                config[name] = value
+
+    @check_auth
+    def do_PATCH_config(self):
+        content_length = int(self.headers.get('content-length', 0))
+        request = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        cluster = self.server.patroni.ha.dcs.get_cluster()
+        data = cluster.config.data.copy()
+        RestApiHandler._patch_config(data, request)
+        if data != cluster.config.data:
+            self.server.patroni.ha.dcs.set_config_value(json.dumps(data, separators=(',', ':')), cluster.config.index)
+            self._write_json_response(200, data)
+        else:
+            self._write_response(304, '', '')
+
+    @check_auth
+    def do_POST_reload(self):
+        try:
+            if not self.server.patroni.config.reload_local_configuration(True):
+                return self._write_response(304, '', '')
```

Comment:
Should we also have some endpoints that one can query to determine if the restart is necessary after the parameter change? Right now, AFAIK one can submit the changes to the configuration but there is no way to ask patroni whether those were applied after the reload....